### ISSUE TITLE
`Style/RedundantFormat` Error

### ISSUE THREAD

**Issue by a-lavis** [rubocop/rubocop#13882]
## Expected behavior

When running `rubocop foo.rb` with the following `.rubocop.yml`:

```yml
AllCops:
  NewCops: enable
  TargetRubyVersion: 3.4
```

and `foo.rb` is the following file:

```ruby
# frozen_string_literal: true

format('(%<foo>?%<bar>s:%<baz>s)', 'foobar')
```

I expect RuboCop to not encounter an error.

## Actual behavior

When running `rubocop foo.rb --debug`, I see the following output:

```
For /Users/aidanlavis/test-rubocop: configuration from /Users/aidanlavis/test-rubocop/.rubocop.yml
Default configuration from /Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/aidanlavis/test-rubocop/foo.rb
An error occurred while Style/RedundantFormat cop was inspecting /Users/aidanlavis/test-rubocop/foo.rb:3:0.
undefined method 'type?' for nil
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/style/redundant_format.rb:152:in 'RuboCop::Cop::Style::RedundantFormat#matching_argument?'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/style/redundant_format.rb:123:in 'block in RuboCop::Cop::Style::RedundantFormat#all_fields_literal?'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/style/redundant_format.rb:118:in 'Array#each'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/style/redundant_format.rb:118:in 'RuboCop::Cop::Style::RedundantFormat#all_fields_literal?'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/style/redundant_format.rb:105:in 'RuboCop::Cop::Style::RedundantFormat#register_all_fields_literal'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/style/redundant_format.rb:101:in 'RuboCop::Cop::Style::RedundantFormat#detect_unnecessary_fields'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/style/redundant_format.rb:83:in 'RuboCop::Cop::Style::RedundantFormat#on_send'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/commissioner.rb:143:in 'Kernel#public_send'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/commissioner.rb:143:in 'block (2 levels) in RuboCop::Cop::Commissioner#trigger_restricted_cops'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/commissioner.rb:171:in 'RuboCop::Cop::Commissioner#with_cop_error_handling'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/commissioner.rb:142:in 'block in RuboCop::Cop::Commissioner#trigger_restricted_cops'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/commissioner.rb:141:in 'Array#each'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/commissioner.rb:141:in 'RuboCop::Cop::Commissioner#trigger_restricted_cops'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/commissioner.rb:70:in 'RuboCop::Cop::Commissioner#on_send'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-ast-1.38.0/lib/rubocop/ast/traversal.rb:20:in 'RuboCop::AST::Traversal#walk'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/commissioner.rb:87:in 'RuboCop::Cop::Commissioner#investigate'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/team.rb:174:in 'RuboCop::Cop::Team#investigate_partial'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cop/team.rb:108:in 'RuboCop::Cop::Team#investigate'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:348:in 'block in RuboCop::Runner#inspect_file'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:347:in 'Array#each'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:347:in 'Enumerable#flat_map'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:347:in 'RuboCop::Runner#inspect_file'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:290:in 'block in RuboCop::Runner#do_inspection_loop'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:324:in 'block in RuboCop::Runner#iterate_until_no_changes'
<internal:kernel>:168:in 'Kernel#loop'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:317:in 'RuboCop::Runner#iterate_until_no_changes'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:286:in 'RuboCop::Runner#do_inspection_loop'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:167:in 'block in RuboCop::Runner#file_offenses'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:192:in 'RuboCop::Runner#file_offense_cache'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:166:in 'RuboCop::Runner#file_offenses'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:154:in 'RuboCop::Runner#process_file'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:135:in 'block in RuboCop::Runner#each_inspected_file'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:134:in 'Array#each'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:134:in 'Enumerable#reduce'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:134:in 'RuboCop::Runner#each_inspected_file'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:120:in 'RuboCop::Runner#inspect_files'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/runner.rb:73:in 'RuboCop::Runner#run'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cli/command/execute_runner.rb:26:in 'block in RuboCop::CLI::Command::ExecuteRunner#execute_runner'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cli/command/execute_runner.rb:52:in 'RuboCop::CLI::Command::ExecuteRunner#with_redirect'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cli/command/execute_runner.rb:25:in 'RuboCop::CLI::Command::ExecuteRunner#execute_runner'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cli/command/execute_runner.rb:17:in 'RuboCop::CLI::Command::ExecuteRunner#run'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cli/command.rb:11:in 'RuboCop::CLI::Command.run'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cli/environment.rb:18:in 'RuboCop::CLI::Environment#run'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cli.rb:122:in 'RuboCop::CLI#run_command'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cli.rb:129:in 'RuboCop::CLI#execute_runners'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cli.rb:51:in 'block in RuboCop::CLI#run'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cli.rb:81:in 'RuboCop::CLI#profile_if_needed'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/lib/rubocop/cli.rb:43:in 'RuboCop::CLI#run'
/Users/aidanlavis/.rbenv/versions/3.4.2/lib/ruby/gems/3.4.0/gems/rubocop-1.72.2/exe/rubocop:19:in '<top (required)>'
/Users/aidanlavis/.rbenv/versions/3.4.2/bin/rubocop:25:in 'Kernel#load'
/Users/aidanlavis/.rbenv/versions/3.4.2/bin/rubocop:25:in '<main>'
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Style/RedundantFormat cop was inspecting /Users/aidanlavis/test-rubocop/foo.rb:3:0.
configuration from /Users/aidanlavis/test-rubocop/.rubocop.yml
Errors are usually caused by RuboCop bugs.
Please, update to the latest RuboCop version if not already in use, and report a bug if the issue still occurs on this version.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.72.2 (using Parser 3.3.7.1, rubocop-ast 1.38.0, analyzing as Ruby 2.7, running on ruby 3.4.2) [arm64-darwin24]
Finished in 0.06480599999849801 seconds
```

## Steps to reproduce the problem

Run `rubocop foo.rb` where `foo.rb` is the above file.

## RuboCop version

```
> rubocop -V
1.72.2 (using Parser 3.3.7.1, rubocop-ast 1.38.0, analyzing as Ruby 3.4, running on ruby 3.4.2) [arm64-darwin24]
```
